salt/utils/network.py

```diff
@@ -1861,7 +1869,7 @@ def dns_check(addr, port, safe=False, ipv6=None):
     family = socket.AF_INET6 if ipv6 else socket.AF_INET if ipv6 is False else socket.AF_UNSPEC
     try:
         refresh_dns()
-        hostnames = socket.getaddrinfo(addr, port, family, socket.SOCK_STREAM)
+        hostnames = socket.getaddrinfo(addr, int(port), family, socket.SOCK_STREAM)
```

Comment:
What's the reasoning for this? Is this just a safety check or is there something specific you're concerned about?

Reply:
I originally added this because of @Ch3LL's [comment](https://github.com/saltstack/salt/issues/46053#issuecomment-366271772) about ensuring port is an int. It felt like an arbitrary and easily solved limitation. However, after looking at the documentation for ``socket.getaddrinfo()``, I realized there are times where someone might want to pass a string (i.e. port='http'), so I reverted the change. Thanks for pointing it out!